src/rbr/RBR.java

```diff
@@ -311,9 +311,8 @@ private void merge(Vertex router) {
      * success
      */
     private boolean mergeUnitary(Vertex router) {
-        for (int a = 0; a < regionsForVertex.get(router).size(); a++) {
-            Region ra = regionsForVertex.get(router).get(a);
-            for (int b = a + 1; b < regionsForVertex.get(router).size(); b++) {
+        for (Region ra : regionsForVertex.get(router)) {
```

Comment:
We cannot use foreach here :disappointed:. This is because the collection is changed during iterations (regions are removed and added) so it can become dangerous to make that change. Sorry.